### PR TITLE
feature/36_LIKE_FUNCTIONALITY

### PR DESCRIPTION
**Pull request by Symphonie234**
- Added a functionality to like listings and show the users or number of users who liked the listing.
- Created new routes in the **web.php** file.
- Created a new migration file for listings with likes
- Created a new model named **ListingLike**

### CODE REVIEW

File: app/Http/Controllers/ListingController.php [hunk 1]
```diff
@@ -139,4 +140,18 @@ public function destroy(Listing $listing)
             return redirect()->route('listings.index')->with('error', 'Failed to delete the listing.');
         }
     }
+
+    public function like(Listing $listing)
```
Comment (ChronoDevs) [inline app/Http/Controllers/ListingController.php:150]: add comment method above

File: app/Http/Controllers/ListingController.php [hunk 2]
```diff
@@ -139,4 +140,18 @@ public function destroy(Listing $listing)
...
+    {
+        $listing->like(auth()->user());
+
+        return redirect()->back();
+    }
+
+    public function unlike(Listing $listing)
```
Comment (ChronoDevs) [inline app/Http/Controllers/ListingController.php:163]: add comment method above

File: resources/views/home/index.blade.php [hunk 3]
```diff
@@ -77,18 +77,41 @@
                     <div class="d-flex justify-content-between align-items-center mt-3">
                         <div>
                             @auth
-                            <!-- Like Link with Icon -->
-                            <a href="#" class=" me-3 action-button">
-                                <i class="far fa-thumbs-up"></i> Like
-                            </a>
+                            <!-- Like Link with Icon and Text -->
+                            @if (!$listing->likes()->where('user_id', auth()->id())->exists())
```
Comment (ChronoDevs) [inline resources/views/home/index.blade.php:81]: don't use query inside html blade

**Comment by Symphonie234** [comment]
- Added comments in the Like and Unlike functions.
- Transferred the query in the **home/index.blade.php** to the **Listing** Model.

**Pull request merged**